### DISCUSSION TITLE
[Question]: Why is the ffmpeg version pinned at n5.1-2, a build from 2022-07-31?

### DISCUSSION THREAD

**Discussion by Aegeontis** [blakeblackshear/frigate#11240] [Ask A Question] [answered]
### What is your question:

Why is the ffmpeg version inside the container image [pinned to a build from 2022-07-31](https://github.com/blakeblackshear/frigate/blob/dev/docker/main/install_deps.sh#L43)?

From my testing the [latest 5.1 build](https://github.com/BtbN/FFmpeg-Builds/releases/download/latest/ffmpeg-n5.1-latest-linux64-gpl-5.1.tar.xz) seems to work fine. I tested it by extracting the latest version and mounting it over `/usr/lib/btbn-ffmpeg` inside the container.

Obviously running the latest-latest version wont work (can confirm). However, running the latest n5.1 version might bring bug fixes and security patches.

Or perhaps anything beyond 2022-07-31 is breaking something on more complicated/different setups than mine?

Sorry if this has already been asked and answered, its rather difficult to find anything related as almost every discussion and issue includes the words "ffmpeg version"

**Comment by NickM-27** [comment, marked as answer]
Because updating ffmpeg is very risky and requires lots of testing. There's no reason to do a small update. We will update to 7.0 in 0.15

**Reply by mateuszdrab** [reply]
I agree, I came across this discussion as I'm currently experimenting with GPU para-virtualization and I'm forced to run a bit older Grid driver version 539.19 on my P40. This is the last version of the driver to support it and the newer datacenter drivers have difficulty switching to WDDM mode.

According to Frigate's documentation on [Hardware Acceleration](https://docs.frigate.video/configuration/hardware_acceleration#nvidia-gpus), FFmpeg requires minimum version 570.0, this used to state 550.0 a few days ago. The link in the documentation points to the current FFmpeg branch.

Realizing that FFmpeg is locked at an older version explains why acceleration is working. 

I think the documentation should be updated to state the current version of FFmpeg, along with it's corresponding Video Codec SDK version and required minimum GPU driver version. I'll try to find out which version it actually requires and post back here.

**Reply by hawkeye217** [reply]
Frigate 0.15 (currently in release candidate status) upgrades ffmpeg to 7.0. See the [release notes](https://github.com/blakeblackshear/frigate/releases/tag/v0.15.0-rc2) and [beta documentation](https://deploy-preview-13787--frigate-docs.netlify.app/).
